core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -192,11 +208,17 @@ private static boolean checkIfLoadCanBeDeleted(LoadMetadataDetails oneLoad,
   }
 
   private static boolean checkIfLoadCanBeDeletedPhysically(LoadMetadataDetails oneLoad,
-      boolean isForceDelete) {
+      boolean isForceDelete, AbsoluteTableIdentifier absoluteTableIdentifier) {
     // Check if the segment is added externally and path is set then do not delete it
     if ((SegmentStatus.MARKED_FOR_DELETE == oneLoad.getSegmentStatus()
-        || SegmentStatus.COMPACTED == oneLoad.getSegmentStatus()) && (oneLoad.getPath() == null
+        || SegmentStatus.COMPACTED == oneLoad.getSegmentStatus() || SegmentStatus
+        .INSERT_IN_PROGRESS == oneLoad.getSegmentStatus()) && (oneLoad.getPath() == null
```

Comment:
better to keep insert_in_progress segment at origin place for a period (for example 3 days?) also.
after expiration days, move it to trash or delete directly.

Reply:
i am not sure about this. maybe we can discuss with  @ajantha-bhat or @akashrn5 once?

Reply:
about clean files, because cleaning different load status have different risk level,
so in my opinion, different load status should have different expiration time for keeping it at the original place.

1. MARKED_FOR_DELETE and COMPACTED => MaxQueryTimeout is enough

2. INSERT_IN_PROGRESS and INSERT_OVERWRITE_IN_PROGRESS => more than data processing time(maybe 3 days)
  if loading or compaction takes a long time which is longer than LockingTimeout and MaxQueryTimeout, clean in_progress will have a big risk.